oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoUtils.java

```diff
@@ -190,4 +195,15 @@ static Type getDocumentStoreExceptionTypeFor(Throwable t) {
         }
         return type;
     }
+
+    /**
+     * Util method to get node size limit for current mongo version
+     *
+     * @param version version of current mongo db
+     * @return size limit based on mongo db version
+     */
+    static int getNodeNameLimit(final String version) {
+        final ServerVersion sv = new ServerVersion(Arrays.stream(version.split("\\.")).map(Integer::new).collect(Collectors.toList()));
+        return sv.compareTo(new ServerVersion(Lists.newArrayList(4,0,0))) > 0 ? Integer.MAX_VALUE : DocumentStore.NODE_NAME_LIMIT;
```

Comment:
I guess you could also do
```suggestion
        return sv.compareTo(new ServerVersion(4, 0) > 0 ? Integer.MAX_VALUE : DocumentStore.NODE_NAME_LIMIT;
```

but that's a detail

Reply:
Hmm, is this really correct? Doesn't this return Integer.MAX_VALUE when e.g. the server is on version 4.0.27?